Generate Certificates guide (#2212)
sensu/sensu-docs: content/sensu-go/5.18/guides/generate-certificates.md

    parent: guides
---

- [Public key infrastructure](#public-key-infrastructure-pki)
- [Issue certificates](#issue-certificates)
    - [Install TLS](#install-tls)
    - [Create a Certificate Authority (CA)](#create-a-certificate-authority-ca)
    - [Generate backend cluster certificates](#generate-backend-cluster-certificates)
    - [Generate agent certificate](#generate-agent-certificate)
- [Next step: Secure Sensu](#next-step-secure-sensu)

If you are deploying Sensu for use outside of a local development environment, you should secure it using transport layer security (TLS).
TLS uses encryption to provide security for communication between Sensu backends and agents as well as communication between human operators and the Sensu backend, such as web UI or sensuctl access.

> We recommend that you configure TLS for your backend clusters from the very beginning because reconfiguring an existing Sensu backend cluster from cleartext to TLS can be time-consuming.

Reviewer: ```suggestion
Because reconfiguring an existing Sensu deployment from cleartext to TLS can be time-consuming, we recommend that you configure TLS for your backend from the very beginning.
```

TLS is also required to use some of Sensu's commercial features, like [secrets management][9] and mutual TLS authentication (mTLS).

This guide explains how to generate the certificates you need to secure Sensu in production.
To use this guide, you must have already [installed Sensu][10] on:

> - Three backend nodes that you plan to cluster together using TLS certificates for transport security.

Reviewer: ```suggestion
- One backend system or three backend systems that you plan to cluster together
```

> - One or more agents that will be configured for mTLS.

Reviewer: ```suggestion
- One or more agents
```

`backend-N-key.pem`    | Backend server private key.  | {{< check >}}       |
`backend-N.csr`        | Certificate signing request. |                     |

> Again, make sure to copy all backend PEM files and `ca-pem` to the corresponding backend system:

Reviewer: ```suggestion
Again, make sure to copy all backend PEM files and `ca.pem` to the corresponding backend system:
```
